### PR TITLE
Various GX matching

### PR DESCRIPTION
**Pull request by r-burns**

### CODE REVIEW

File: src/dolphin/gx/GXTransform.c [hunk 1]
```diff
@@ -114,128 +111,67 @@ asm void GXProject(f32 x, f32 y, f32 z, MtxPtr mtx, f32* pm, f32* vp, f32* sx, f
...
+    __GXContexts.main->projection_v[4] = proj[10];
+    __GXContexts.main->projection_v[5] = proj[11];
+    if ((s32) projection_type == 1) {
+        __GXContexts.main->projection_v[1] = proj[3];
+        __GXContexts.main->projection_v[3] = proj[7];
+    } else {
+        __GXContexts.main->projection_v[1] = proj[2];
+        __GXContexts.main->projection_v[3] = proj[6];
+    }
+    WGPIPE.u8 = 0x10;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:128]:
```suggestion
    WGPIPE.u8 = GX_LOAD_XF_REG;
```

File: src/dolphin/gx/GXTransform.c [hunk 2]
```diff
@@ -114,128 +111,67 @@ asm void GXProject(f32 x, f32 y, f32 z, MtxPtr mtx, f32* pm, f32* vp, f32* sx, f
...
+    WGPIPE.u32 = 0x61020;
+    WGPIPE.f32 = __GXContexts.main->projection_v[0];
+    WGPIPE.f32 = __GXContexts.main->projection_v[1];
+    WGPIPE.f32 = __GXContexts.main->projection_v[2];
+    WGPIPE.f32 = __GXContexts.main->projection_v[3];
+    WGPIPE.f32 = __GXContexts.main->projection_v[4];
+    WGPIPE.f32 = __GXContexts.main->projection_v[5];
+    WGPIPE.u32 = __GXContexts.main->projection_type;
+    __GXContexts.main->x0.u16[1] = 1;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:137]:
```suggestion
    set_x2(GX_TRUE);
```

File: src/dolphin/gx/GXTransform.c [hunk 3]
```diff
@@ -591,63 +391,30 @@ asm unk_t GXSetScissorBoxOffset()
...
-/* 80341864 0033E444  38 00 00 01 */	li r0, 1
-/* 80341868 0033E448  90 A6 80 00 */	stw r5, -0x8000(r6)
-/* 8034186C 0033E44C  90 66 80 00 */	stw r3, -0x8000(r6)
-/* 80341870 0033E450  B0 04 00 02 */	sth r0, 2(r4)
-/* 80341874 0033E454  4E 80 00 20 */	blr 
-} // clang-format on
-#pragma pop
+void GXSetClipMode(s32 mode)
+{
+    WGPIPE.u8 = 0x10;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:396]:
```suggestion
    WGPIPE.u8 = GX_LOAD_XF_REG;
```

File: src/dolphin/gx/GXTransform.c [hunk 4]
```diff
@@ -591,63 +391,30 @@ asm unk_t GXSetScissorBoxOffset()
...
+    WGPIPE.u32 = 0x1005;
+    WGPIPE.u32 = mode;
+    __GXContexts.main->x0.u16[1] = 1;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:399]:
```suggestion
    set_x2(GX_TRUE);
```

File: src/dolphin/gx/GXTransform.c [hunk 5]
```diff
@@ -591,63 +391,30 @@ asm unk_t GXSetScissorBoxOffset()
...
+    if (arg0 < 5) {
+        WGPIPE.u8 = 8;
+        WGPIPE.u8 = 0x30;
+        WGPIPE.u32 = __GXContexts.main->x80;
+        WGPIPE.u8 = 0x10;
+        WGPIPE.u32 = 0x1018;
+        WGPIPE.u32 = __GXContexts.main->x80;
+    } else {
+        WGPIPE.u8 = 8;
+        WGPIPE.u8 = 0x40;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:413]:
```suggestion
        WGPIPE.u8 = GX_CMD_CALL_DL;
```

File: src/dolphin/gx/GXTransform.c [hunk 6]
```diff
@@ -591,63 +391,30 @@ asm unk_t GXSetScissorBoxOffset()
...
+        WGPIPE.u32 = __GXContexts.main->x84;
+        WGPIPE.u8 = 0x10;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:415]:
```suggestion
        WGPIPE.u8 = GX_LOAD_XF_REG;
```

File: src/dolphin/gx/GXTransform.c [hunk 7]
```diff
@@ -591,63 +391,30 @@ asm unk_t GXSetScissorBoxOffset()
...
+        WGPIPE.u32 = 0x1019;
+        WGPIPE.u32 = __GXContexts.main->x84;
+    }
+    __GXContexts.main->x0.u16[1] = 1;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTransform.c:419]:
```suggestion
    set_x2(GX_TRUE);
```

File: src/dolphin/gx/GX_unknown_001.h [hunk 8]
```diff
@@ -8,11 +8,11 @@
 /* 003FDDC0 */ extern const GXRenderModeObj lbl_804011E0;
 /* 003FDD84 */ extern const GXRenderModeObj lbl_804011A4;
 /* 003FDD48 */ extern const GXRenderModeObj GXNtsc480IntDf;
-/* 0033E33C */ extern UNK_FUNC(GXSetScissor);
```
Comment (ribbanya) [inline src/dolphin/gx/GX_unknown_001.h:11]: Not a fan of this macro, I take it.

File: src/dolphin/gx/GXTexture.c [hunk 9]
```diff
@@ -1055,31 +1042,15 @@ void GXInitTlutRegion(s32* arg0, s32 arg1, s32 arg2)
...
-/* 8033F2B4 0033BE94  4E 80 00 20 */	blr 
-} // clang-format on
-#pragma pop
+void GXInvalidateTexAll(void)
+{
+    __GXFlushTextureState();
+    WGPIPE.u8 = 0x61;
+    WGPIPE.u32 = 0x66001000;
+    WGPIPE.u8 = 0x61;
+    WGPIPE.u32 = 0x66001100;
```
Comment (ribbanya) [inline src/dolphin/gx/GXTexture.c:1051]:
```suggestion
    WGPIPE.u8 = GX_LOAD_BP_REG;
    WGPIPE.u32 = 0x66001000;
    WGPIPE.u8 = GX_LOAD_BP_REG;
    WGPIPE.u32 = 0x66001100;
```

**Review by ribbanya — CHANGES_REQUESTED** [review]
A lot of this requires cleanup, but that applies to the rest of this module so it's fine for now.

File: src/dolphin/gx/GX_unknown_001.h (hunk 8 above)
Comment (r-burns) [reply src/dolphin/gx/GX_unknown_001.h:11]: Nah it's fine. This change is because the func isn't unk anymore.

**Review by ribbanya — APPROVED** [review]

**Pull request merged**